flow/layers/layer.h

```diff
@@ -80,12 +89,54 @@ struct PrerollContext {
   // than to remember the value so that it can choose the right strategy
   // for its |Paint| method.
   bool subtree_can_inherit_opacity = false;
+
+  std::vector<RasterCacheEntry> raster_cached_entries;
 };
 
-class PictureLayer;
-class DisplayListLayer;
-class PerformanceOverlayLayer;
-class TextureLayer;
+struct RasterCacheEntry {
+  RasterCacheEntry(CacheableLayer* layer,
+                   PrerollContext& context,
+                   const SkMatrix& matrix,
+                   unsigned num_child,
+                   bool need_caching = false)
+      : layer(layer),
+        matrix(matrix),
+        raster_cache(context.raster_cache),
+        mutators_stack(context.mutators_stack),
+        cull_rect(context.cull_rect),
+        color_space(context.dst_color_space),
+        num_child_entries(num_child),
+        need_caching(need_caching) {}
+
+  explicit RasterCacheEntry(CacheableLayer* layer,
+                            PrerollContext& context,
+                            const SkMatrix& matrix,
+                            bool need_caching = false)
+      : RasterCacheEntry(layer, context, matrix, 0, need_caching) {}
+
+  void UpdateRasterCacheEntry(CacheableLayer* layer,
+                              PrerollContext& context,
+                              const SkMatrix& matrix) {
+    this->raster_cache = context.raster_cache;
+    this->layer = layer;
+    this->matrix = matrix;
+    this->mutators_stack = context.mutators_stack;
+    this->cull_rect = context.cull_rect;
+    this->color_space = context.dst_color_space;
+    this->surface_needs_readback = context.surface_needs_readback;
+  }
+
+  CacheableLayer* layer;
+  SkMatrix matrix;
+  RasterCache* raster_cache;
+  MutatorsStack mutators_stack;
+  SkRect cull_rect;
+  SkColorSpace* color_space;
+
+  bool surface_needs_readback;
```

Comment:
Generally this value is only used during Preroll and will not be needed in the caching methods. You can delete this field and just provide a dummy value when creating the Context used when processing the caching queue.